### PR TITLE
[doc][state observability] Updated exception/doc for state api when dashboard not available

### PR DESCRIPTION
**Pull request by rickyyx**
Signed-off-by: rickyyx <rickyx@anyscale.com>

## Why are these changes needed?

- Updated doc/exception string to reflect requirement of dashboard 
- Wrap state api exception in click exception for CLI usage to hide stacktrace. 
Now
```

(dev) ➜  ray git:(master) ✗ ray list nodes
Usage: ray list [OPTIONS] [actors|jobs|placement-
                groups|nodes|workers|tasks|objects|runtime-envs|cluster-
                events]

Error: Failed to make request to http://127.0.0.1:8265/api/v0/nodes. Failed to connect to API server. Please check the API server log for details. Make sure dependencies are installed with `pip install ray[default]`. Please also check dashboard is available, and included when starting ray cluster, i.e. `include_dashboard=True`.Response(url=http://127.0.0.1:8265/api/v0/nodes?limit=100&timeout=24&_server_timeout_multiplier=0.8,status=404)
==========ALPHA, FEEDBACK NEEDED ===============
State Observability APIs is currently in Alpha. 
If you have any feedback, you could do so at either way as below:
    1. Report bugs/issues with details: https://forms.gle/gh77mwjEskjhN8G46
    2. Follow up in #ray-state-observability-dogfooding slack channel of Ray: https://tinyurl.com/2pm26m4a
==========================================================
```

Before
```
(dev) ➜  ray git:(master) ✗ ray list nodes
Traceback (most recent call last):
  File "/data/home/rickyx/ray/python/ray/experimental/state/api.py", line 202, in _make_http_get_request
    response.raise_for_status()
  File "/data/home/rickyx/anaconda3/envs/dev/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://127.0.0.1:8265/api/v0/nodes?limit=100&timeout=24&_server_timeout_multiplier=0.8

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/data/home/rickyx/anaconda3/envs/dev/bin/ray", line 33, in <module>
    sys.exit(load_entry_point('ray', 'console_scripts', 'ray')())
  File "/data/home/rickyx/ray/python/ray/scripts/scripts.py", line 2596, in main
    return cli()
  File "/data/home/rickyx/anaconda3/envs/dev/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/data/home/rickyx/anaconda3/envs/dev/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/data/home/rickyx/anaconda3/envs/dev/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/data/home/rickyx/anaconda3/envs/dev/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/data/home/rickyx/anaconda3/envs/dev/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/data/home/rickyx/ray/python/ray/experimental/state/state_cli.py", line 536, in ray_list
    data = client.list(
  File "/data/home/rickyx/ray/python/ray/experimental/state/api.py", line 466, in list
    list_api_response = self._make_http_get_request(
  File "/data/home/rickyx/ray/python/ray/experimental/state/api.py", line 220, in _make_http_get_request
    raise RayStateApiException(err_str) from e
ray.experimental.state.exception.RayStateApiException: Failed to make request to http://127.0.0.1:8265/api/v0/nodes. Response(url=http://127.0.0.1:8265/api/v0/nodes?limit=100&timeout=24&_server_timeout_multiplier=0.8,status=404)
==========ALPHA, FEEDBACK NEEDED ===============
State Observability APIs is currently in Alpha. 
If you have any feedback, you could do so at either way as below:
    1. Report bugs/issues with details: https://forms.gle/gh77mwjEskjhN8G46
    2. Follow up in #ray-state-observability-dogfooding slack channel of Ray: https://tinyurl.com/2pm26m4a
==========================================================
```

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/experimental/state/api.py [hunk 1]
```diff
@@ -200,23 +200,22 @@ def _make_http_get_request(
...
-                        "Failed to connect to API server. Please check the API server "
-                        "log for details. Make sure dependencies are installed with "
-                        "`pip install ray[default]`."
-                    )
+                err_str += (
+                    "Failed to connect to API server. Please check the API server "
+                    "log for details. Make sure dependencies are installed with "
+                    "`pip install ray[default]`. Please also check dashboard is "
+                    "available, and included when starting ray cluster, "
+                    "i.e. `include_dashboard=True`."
```
Comment (rkooo567) [inline python/ray/experimental/state/api.py:213]:
```suggestion
                    "i.e. `include_dashboard=True`. "
```

File: python/ray/experimental/state/api.py [hunk 2]
```diff
@@ -200,23 +200,22 @@ def _make_http_get_request(
...
 
                 # Best-effort to give hints to users on potential reasons of connection
                 # failure.
-                if isinstance(e, requests.exceptions.ConnectionError):
-                    err_str += (
-                        "Failed to connect to API server. Please check the API server "
-                        "log for details. Make sure dependencies are installed with "
-                        "`pip install ray[default]`."
-                    )
+                err_str += (
```
Comment (rkooo567) [inline python/ray/experimental/state/api.py:208]: any reason why we remove ConnectionError?

**Review by rkooo567 — APPROVED** [review]

File: python/ray/experimental/state/api.py (hunk 1 above)
Comment (scottsun94) [inline python/ray/experimental/state/api.py:213]: Maybe make the example more complete like, "i.e. `ray start --include_dashboard=True`."

**Review by scottsun94 — APPROVED** [review]

File: python/ray/experimental/state/api.py (hunk 2 above)
Comment (rickyyx) [reply python/ray/experimental/state/api.py:208]: ConnectionError is a [derived class](https://requests.readthedocs.io/en/v3.0.0/_modules/requests/exceptions/) of that `requests.exceptions.RequestException` - so I am catching more requests related exceptions, which I feel should be handled similarly (i.e. a request to server failure)

**Review by maxpumperla — APPROVED** [review]

**Review by rkooo567 — COMMENTED** [review]
One last comment!

**Review by scottsun94 — APPROVED** [review]
LGTM. Thanks!

**Comment by rickyyx** [comment]
Doc: https://ray--29390.org.readthedocs.build/en/29390/ray-observability/state/state-api.html

**Comment by rkooo567** [comment]
don't forget to cherry pick!

**Pull request merged**